### ISSUE TITLE
Data ingestion fails on Quantiles DoublesSketch containing NULL When usingDefaultValueForNull = false

### ISSUE THREAD

**Issue by hnakamor** [apache/incubator-druid#7108]
Hi All,

In 0.13.0-incubating, setting usingDefaultValueForNull = false and using a quantilesDoubleSketch aggregator at ingestion containing NULL will result in the following exception during index task:

```
2019-02-18T10:11:33,248 ERROR [task-runner-0-priority-0] org.apache.druid.indexing.common.task.IndexTask - Encountered exception in BUILD_SEGMENTS.
java.lang.StringIndexOutOfBoundsException: String index out of range: 0
    at java.lang.String.charAt(String.java:658) ~[?:1.8.0_191]
    at org.apache.druid.query.aggregation.datasketches.quantiles.DoublesSketchComplexMetricSerde$1.extractValue(DoublesSketchComplexMetricSerde.java:78) ~[?:?]
    at org.apache.druid.segment.incremental.IncrementalIndex$1IncrementalIndexInputRowColumnSelectorFactory$1.getObject(IncrementalIndex.java:197) ~[druid-processing-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.query.aggregation.datasketches.quantiles.DoublesSketchMergeAggregator.aggregate(DoublesSketchMergeAggregator.java:42) ~[?:?]
    at org.apache.druid.segment.incremental.OnheapIncrementalIndex.doAggregate(OnheapIncrementalIndex.java:253) ~[druid-processing-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.segment.incremental.OnheapIncrementalIndex.addToFacts(OnheapIncrementalIndex.java:167) ~[druid-processing-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.segment.incremental.IncrementalIndex.add(IncrementalIndex.java:610) ~[druid-processing-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.segment.realtime.plumber.Sink.add(Sink.java:181) ~[druid-server-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.segment.realtime.appenderator.AppenderatorImpl.add(AppenderatorImpl.java:246) ~[druid-server-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.segment.realtime.appenderator.BaseAppenderatorDriver.append(BaseAppenderatorDriver.java:403) ~[druid-server-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.segment.realtime.appenderator.BatchAppenderatorDriver.add(BatchAppenderatorDriver.java:106) ~[druid-server-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.indexing.common.task.IndexTask.generateAndPublishSegments(IndexTask.java:1032) ~[druid-indexing-service-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.indexing.common.task.IndexTask.run(IndexTask.java:466) [druid-indexing-service-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner$SingleTaskBackgroundRunnerCallable.call(SingleTaskBackgroundRunner.java:421) [druid-indexing-service-0.13.0-incubating.jar:0.13.0-incubating]
    at org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner$SingleTaskBackgroundRunnerCallable.call(SingleTaskBackgroundRunner.java:393) [druid-indexing-service-0.13.0-incubating.jar:0.13.0-incubating]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_191]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_191]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_191]
    at java.lang.Thread.run(Thread.java:748) [?:1.8.0_191]
```

Ingest data:

```
1549929600000	1		1
1549929600000	2		1
1549929600000	3	0.332583716	1
1549929600000	4	0.47188857	1
1549929700000	1		1
1549929700000	2	0.39832398	1
1549929700000	3		1
1549929700000	4	0.366082674	1
```

Ingest Setting:

```
      "parser" : {
        "type" : "string",
        "parseSpec" : {
          "format" : "tsv",
          "columns": ["timestamp", "index", "value", "extra"],
          "timestampSpec": {
            "column": "timestamp",
            "format": "millis"
          },
          "dimensionsSpec" : {
            "dimensions" : ["index"]
          }
        }
      },
      "metricsSpec" : [
        {"type": "quantilesDoublesSketch", "name": "sketch", "fieldName": "value", "k": 128},
        {"type": "count", "name": "count"}
      ],
```

**Comment by gianm** [comment]
Should be fixed by #7429. Thanks for the report!

**Comment by hnakamor** [comment]
@gianm Thank you for fixing this bug! I'm happy to contribute a little.
